File: src/com/manbot/service/Service.java
package com.manbot.service;

import java.util.concurrent.TimeUnit;

public abstract class Service implements Runnable
{
    public final long initialDelay;
    public final long intervalDelay;
    public final TimeUnit timeUnit;

    protected Service(long initialDelaySeconds, long intervalDelaySeconds)
    {
        this(initialDelaySeconds, intervalDelaySeconds, TimeUnit.SECONDS);
    }

    protected Service(long initialDelay, long intervalDelay, TimeUnit unit)
    {
        this.initialDelay = initialDelay;
        this.intervalDelay = intervalDelay;
        this.timeUnit = unit;
    }
}

File: bot/src/main/kotlin/com/manbot/task/Service.kt
package com.manbot.task

import com.manbot.Manbot
import java.util.concurrent.TimeUnit

/**
 * @author Pavan C. (pavan407)
 */
abstract class Service @JvmOverloads constructor(
        val initialDelay: Long = 0,
        val intervalDelay: Long,
        val timeUnit: TimeUnit = TimeUnit.SECONDS
) : Task()
{
    override fun onInit()
    {
        Manbot.scheduleService(this)
    }
}
